### PR TITLE
Version 4.1

### LINKED ISSUE #498: Deprecation of `setup.py install` when `wheel` is absent
#### Description
The legacy `setup.py install` method should be removed from the installation methods.
Follow the suggestions provided at https://github.com/pypa/pip/issues/8559.

### LINKED ISSUE #522: Modern issue templates
#### Description
Updating repo issue templates

Reference:
https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository

### LINKED ISSUE #523: Python 3.12 support
#### Description
Python 3.12 Support

### LINKED ISSUE #524: New diagrams
#### Description
Updating block diagrams

### PR DESCRIPTION
**Pull request by sepandhaghighi**
#### Reference Issues/PRs
fix #498 
fix #522 
fix #523 
fix #524

#### What does this implement/fix? Explain your changes.
Version 4.1

### CODE REVIEW

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
minor changes.

File: README.md
Comment (alirezazolanvari) [inline README.md:1]: Please add DO as one of the project's supporters.

Comment (sepandhaghighi) [reply README.md:1]: fixed in https://github.com/sepandhaghighi/pycm/pull/563/commits/36bbf9718a792ee8da13c6749b0f4adb5645326b

**Review by alirezazolanvari — APPROVED** [review]

File: Document/Example1.ipynb [hunk 1]
```diff
@@ -61,7 +61,15 @@
...
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Keyring is skipped due to an exception: invalid syntax (core.py, line 48)\n"
+     ]
+    }
+   ],
```
[the remark this answers is not in the document]
Comment (sepandhaghighi) [reply Document/Example1.ipynb:72]: It's not related to the PyCM code. I think this warning is associated with the pip version.

**Review by sadrasabouri — APPROVED** [review]

**Pull request merged**